spring-kafka/src/main/java/org/springframework/kafka/retrytopic/DefaultDestinationTopicResolver.java

```diff
@@ -58,18 +59,31 @@
 
 	private final Map<String, DestinationTopic> destinationsTopicMap;
 
-	private final Clock clock;
+	private Clock clock;
 
-	private final ApplicationContext applicationContext;
+	private ApplicationContext applicationContext;
 
 	private boolean contextRefreshed;
 
-	public DefaultDestinationTopicResolver(Clock clock, ApplicationContext applicationContext) {
-		this.applicationContext = applicationContext;
-		this.clock = clock;
+	/**
+	 * Constructs a new instance.
+	 * The default {@link Clock} can be overridden via the
+	 * {@link #setClock(Clock)} method.
+	 * @see RetryTopicBootstrapper
+	 */
+	public DefaultDestinationTopicResolver() {
 		this.sourceDestinationsHolderMap = new HashMap<>();
 		this.destinationsTopicMap = new HashMap<>();
 		this.contextRefreshed = false;
+		this.clock = Clock.systemUTC();
+	}
+
+	@SuppressWarnings("deprecated")
```

Comment:
Why do we need this suppression here?
Doesn't look like you use some other deprecated API...

Reply:
I thought my IDE was complaining, I'll remove it, thanks.